lib/msf/core/exploit/sqli/mysqli/common.rb

```diff
@@ -213,10 +213,11 @@ def write_to_file(fpath, data)
     #
     # Attempt reading from a file on the filesystem, requires having the FILE privilege
     # @param fpath [String] The path of the file to read
+    # @param binary [Boolean] Whether the target file is a binary one or not
     # @return [String] The content of the file if reading was successful
     #
-    def read_from_file(fpath)
-      run_sql("select load_file('#{fpath}')")
+    def read_from_file(fpath, binary=false)
+      call_function("load_file('#{fpath}')")
```

Comment:
Won't this be an issue if the path contains single quotes at all? You can create paths on both Linux and Windows with single quotes so I'd advise looking at escaping the path before putting it into something like this.

This also applies to the other cases here.

Reply:
I'm looking for a way to fix it, not allowing quotes at all seems to be a bad idea (as they might be escaped, and part of filenames on Linux, they might not be escaped, and valid on Windows). But I have yet to find a way to check if a path is valid. I'll look for path specifications before trying to address this.